server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageFactory.java

```diff
@@ -110,14 +111,24 @@ public Message fromMetaDataWithContent(MetaDataWithContent message) throws Mailb
                 .bcc(fromAddressList(mimeMessage.getBcc()))
                 .replyTo(fromAddressList(mimeMessage.getReplyTo()))
                 .size(message.getSize())
-                .date(message.getInternalDateAsZonedDateTime())
+                .date(getDateFromHeaderOrInternalDateOtherwise(mimeMessage, message))
                 .textBody(textBody)
                 .htmlBody(htmlBody)
                 .preview(preview)
                 .attachments(getAttachments(message.getAttachments()))
                 .build();
     }
 
+    private ZonedDateTime dateToZonedDate(Date date) {
+        return ZonedDateTime.ofInstant(date.toInstant(), UTC_ZONE_ID);
```

Comment:
This is time zone dependant. Depending of the timezone I'm runnning in, I will get a different result.